### PR TITLE
LPS-165980, LPS-165981 | Test Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
[LPS-165980](https://issues.liferay.com/browse/LPS-165980)
[LPS-165981](https://issues.liferay.com/browse/LPS-165981)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -2142,6 +2142,93 @@ definition {
...
+				itemKey = "PicklistItemTest",
+				itemName = "Picklist Item Test",
+				picklistName = "Picklist Test");
+		}
+
+		task ("Add a Custom Object and Custom Picklist (via API)") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 161898",
+				objectName = "CustomObject161898",
+				pluralLabelName = "Custom Objects 161898");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2248]: Use the ticket number correct to add the Custom Object - [LPS-165980](https://issues.liferay.com/browse/LPS-165980)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -2142,6 +2142,93 @@ definition {
...
+				filterBy = "Field Picklist",
+				filterValue = "Picklist Item Test");
+
+			Button.clickSave();
+		}
+
+		task ("Assert that the filter option is on Custom Object") {
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject161898");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject161898");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2305]: Also too - [LPS-165980](https://issues.liferay.com/browse/LPS-165980)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 3]
```diff
@@ -2142,6 +2142,93 @@ definition {
...
+			ObjectCustomViews.addNewFilterViaUI(
+				filterBy = "Field Picklist",
+				filterType = "Includes",
+				filterValue = "Picklist Item Test");
+
+			ObjectCustomViews.assertPresentEntriesOnFilters(
+				filterBy = "Field Picklist",
+				filterValue = "Picklist Item Test");
+
+			Button.clickSave();
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2299]: Change `Button.clickSave();` to `PortletEntry.save();`

> The PortletEntry macros wait for JS events to fire before moving forward in the test. Without waiting, test assets sometimes aren't added properly leading to failed assertions.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 4]
```diff
@@ -2106,6 +2106,93 @@ definition {
...
+				itemKey = "PicklistItemTest",
+				itemName = "Picklist Item Test",
+				picklistName = "Picklist Test");
+		}
+
+		task ("Add a Custom Object and Custom Picklist (via API)") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 161899",
+				objectName = "CustomObject161899",
+				pluralLabelName = "Custom Objects 161899");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2125]: Use the ticket number correct to add the Custom Object - [LPS-165981](https://issues.liferay.com/browse/LPS-165981)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 5]
```diff
@@ -2106,6 +2106,93 @@ definition {
...
+		}
+
+		task ("Go to Custom Object and create a Custom Picklist Field") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 161899");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2131]: Also too - [LPS-165981](https://issues.liferay.com/browse/LPS-165981)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 6]
```diff
@@ -2106,6 +2106,93 @@ definition {
...
+				filterBy = "Field Picklist",
+				filterValue = "Picklist Item Test");
+
+			Button.clickSave();
+		}
+
+		task ("Assert that the filter option is on Custom Object") {
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject161899");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject161899");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2182]: Also too - [LPS-165981](https://issues.liferay.com/browse/LPS-165981)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 7]
```diff
@@ -2142,6 +2142,93 @@ definition {
...
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 161898",
+				objectName = "CustomObject161898",
+				pluralLabelName = "Custom Objects 161898");
+		}
+
+		task ("Go to Custom Object and create a Custom Picklist Field") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 161898");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2254]: Also too - [LPS-165980](https://issues.liferay.com/browse/LPS-165980)

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 8]
```diff
@@ -2106,6 +2106,93 @@ definition {
...
+			ObjectCustomViews.addNewFilterViaUI(
+				filterBy = "Field Picklist",
+				filterType = "Excludes",
+				filterValue = "Picklist Item Test");
+
+			ObjectCustomViews.assertPresentEntriesOnFilters(
+				filterBy = "Field Picklist",
+				filterValue = "Picklist Item Test");
+
+			Button.clickSave();
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2176]: Change `Button.clickSave();` to `PortletEntry.save();`

> The PortletEntry macros wait for JS events to fire before moving forward in the test. Without waiting, test assets sometimes aren't added properly leading to failed assertions.

**Review by andreefarias — CHANGES_REQUESTED** [review]
Hi @rafaellajordao,

The testcases were well written. Added some comments to avoid CI errors. Check and fix it. Thanks!

**Comment by rafaellajordao** [comment]
conflict resolved in this https://github.com/liferay-bpm-qa/liferay-portal/pull/279

**Pull request closed without merging**